### PR TITLE
add h101 PCB

### PR DESCRIPTION
**Pull request by hineybush**
## Description

Add h101 PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hineybush/h101/config.h
Comment (zvecr) [inline keyboards/hineybush/h101/config.h:1]: This file should be removed.

File: keyboards/hineybush/h101/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Default h101 keymap, using a WKL/Tsangan bottom row, 2u backspace, 2u numpad +/enter, no short shift.
```
Comment (zvecr) [inline keyboards/hineybush/h101/keymaps/default/keymap.c:1]:
```suggestion
// Copyright 2023 Josh Hinnebusch (@hineybush)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/hineybush/h101/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Default VIA keymap for h101, using the "all" layout
+
```
Comment (zvecr) [inline keyboards/hineybush/h101/keymaps/via/keymap.c:2]:
```suggestion
// Copyright 2023 Josh Hinnebusch (@hineybush)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/hineybush/h101/config.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+#define LED_NUM_LOCK_PIN A6
+#define LED_CAPS_LOCK_PIN B14
+#define LED_SCROLL_LOCK_PIN B8
+#define LED_ON_STATE 0
```
Comment (fauxpark) [inline keyboards/hineybush/h101/config.h:25]: These can go in info.json, and this file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/hineybush/h101/keymaps/via_F12/keymap.c
[the remark this answers is not in the document]
Comment (hineybush) [reply keyboards/hineybush/h101/keymaps/via_F12/keymap.c:1]: the maps need to be different due to the matrix assignments, I tried it with the way you described and it didn't work properly.

File: keyboards/hineybush/h101/info.json [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+        "pid": "0x0002"
+    },
+    "indicators": {
+        "num_lock": "A6",
+        "caps_lock": "B14",
+        "scroll_lock": "B8",
+        "on_state": 0
+    },
+    "layouts": {
+        "LAYOUT_h101_tsangan": {
```
Comment (waffle87) [inline keyboards/hineybush/h101/info.json:33]: Layout names shouldn't contain the keyboard name.
```suggestion
        "LAYOUT_tsangan": {
```

File: keyboards/hineybush/h101/info.json [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+                {"matrix": [5,12], "label":"Win", "x":12.5, "y":5.25},
+                {"matrix": [5,13], "label":"Ctrl", "x":13.5, "y":5.25, "w":1.5},
+                {"matrix": [5,15], "label":"Left Arrow", "x":15.25, "y":5.25},
+                {"matrix": [5,16], "label":"Down Arrow", "x":16.25, "y":5.25},
+                {"matrix": [5,17], "label":"Right Arrow", "x":17.25, "y":5.25},
+                {"matrix": [5,19], "label":"0", "x":18.5, "y":5.25, "w":2},
+                {"matrix": [5,20], "label":".", "x":20.5, "y":5.25}
+            ]
+        },
+        "LAYOUT_h101_all": {
```
Comment (waffle87) [inline keyboards/hineybush/h101/info.json:145]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/hineybush/h101/info.json [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+                {"matrix": [5,15], "label":"Left Arrow", "x":15.25, "y":5.25},
+                {"matrix": [5,16], "label":"Down Arrow", "x":16.25, "y":5.25},
+                {"matrix": [5,17], "label":"Right Arrow", "x":17.25, "y":5.25},
+                {"matrix": [5,18], "label":"0", "x":18.5, "y":5.25},
+                {"matrix": [5,19], "label":"0", "x":19.5, "y":5.25},
+                {"matrix": [5,20], "label":".", "x":20.5, "y":5.25},
+                {"matrix": [5,21], "label":"Enter", "x":21.5, "y":5.25}
+            ]
+        },
+        "LAYOUT_h100_all": {
```
Comment (waffle87) [inline keyboards/hineybush/h101/info.json:263]: And here, if this layout ends up staying.

File: keyboards/hineybush/h101/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,                                                  KC_UP,                       KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+    KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_RALT, MO(1),   KC_RCTL,                                                                                       KC_LEFT, KC_DOWN, KC_RGHT,            KC_P0,   KC_PDOT                   ),
+
+  [1] = LAYOUT_h101_tsangan(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS ,KC_TRNS, KC_TRNS, KC_TRNS,                        KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                        KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                        KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
```
Comment (waffle87) [inline keyboards/hineybush/h101/keymaps/default/keymap.c:18]:
```suggestion
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                        KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
```

File: keyboards/hineybush/h101/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_RALT, KC_LGUI, MO(1),   KC_RCTL,                                                                                       KC_LEFT, KC_DOWN, KC_RGHT,            KC_TRNS, KC_P0,   KC_PDOT, KC_TRNS),
+
+  [1] = LAYOUT_h101_all(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS ,KC_TRNS, KC_TRNS, KC_TRNS,                                 KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                        KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                 KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
```
Comment (waffle87) [inline keyboards/hineybush/h101/keymaps/via/keymap.c:18]:
```suggestion
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                 KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
```

File: keyboards/hineybush/h101/readme.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```
Comment (waffle87) [inline keyboards/hineybush/h101/readme.md:27]:
```suggestion
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/hineybush/h101/keymaps/via_F12/keymap.c
Comment (drashna) [inline keyboards/hineybush/h101/keymaps/via_F12/keymap.c:1]: could you rename the folder to be lower case? 

Also, in theory, the "all" layout should cover this.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]
Looks like you accidentally pulled in some changes to the submodules.  Those changes need to be reverted.

**Comment by drashna** [comment]
```sh
cd lib/lvgl
git checkout e19410f8f
cd ../pico-sdk
git checkout a3398d8
```

And then commit, and push that commit.  That should fix the issue.

**Comment by dunk2k** [comment]
As their multiple layout options for this PCB, could a `matrix_diagram.md` be provided?

File: keyboards/hineybush/h101/keymaps/via_F12/keymap.c
[the remark this answers is not in the document]
Comment (zvecr) [reply keyboards/hineybush/h101/keymaps/via_F12/keymap.c:1]: Depends what you mean by "didnt work". If its that some matrix positions were not filled in with default values, then the issue is your `LAYOUT_all` does not represent all matrix locations. Other its difficult to guess, but likely an issue with the via config.

Duplicating the via keymap does little to resolve the root of the issue and shouldnt be accepted to the repo.

**Comment by hineybush** [comment]
> As their multiple layout options for this PCB, could a `matrix_diagram.md` be added? template provided:
> 
> ```
> ┌───┐┌───┬───┬───┬───┐┌───┬───┬───┬───┐┌───┬───┬───┬───┐┌───┐ ┌───┬───┬───┐ ┌───┬───┬───┬───┐
> │   ││   │   │   │   ││   │   │   │   ││   │   │   │   ││   │ │   │   │   │ │   │   │   │   │
> └───┘└───┴───┴───┴───┘└───┴───┴───┴───┘└───┴───┴───┴───┘└───┘ └───┴───┴───┘ └───┴───┴───┴───┘
> ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐ ┌───┬───┬───┐ ┌───┬───┬───┬───┐   ┌───────┐
> │   │   │   │   │   │   │   │   │   │   │   │   │   │   │   │ │   │   │   │ │   │   │   │   │   │       │ 2u Backspace
> ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┤ ├───┼───┼───┤ ├───┼───┼───┼───┤   └───────┘                ┌───┐
> │     │   │   │   │   │   │   │   │   │   │   │   │   │     │ │   │   │   │ │   │   │   │   │                            │   │
> ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤ └───┴───┴───┘ ├───┼───┼───┼───┤                            │   │ Numpad "+"
> │      │   │   │   │   │   │   │   │   │   │   │   │        │               │   │   │   │   │                            │   │
> ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┤     ┌───┐     ├───┼───┼───┼───┤                            ├───┤
> │        │   │   │   │   │   │   │   │   │   │   │      │   │     │   │     │   │   │   │   │                            │   │
> ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬┴───┤ ┌───┼───┼───┐ ├───┼───┼───┼───┤                ┌───────┐   │   │ Numpad Enter
> │    │    │    │                        │    │    │    │    │ │   │   │   │ │   │   │   │   │       Numpad 0 │       │   │   │
> └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘ └───┴───┴───┘ └───┴───┴───┴───┘                └───────┘   └───┘
>                                                  ┌──────────┐
>                                     2.75u RShift │          │
>                                                  └──────────┘
> ┌─────┬───┬─────┬───────────────────────────┬─────┬───┬─────┐
> │     │   │     │                           │     │   │     │ Tsangan/WKL
> └─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
> ```

I don't think that I've ever added this in for any of my PCB firmware, I'd prefer to keep it that way

**Review by drashna — APPROVED** [review]
I have no preference about the matrix diagram getting added here.  It will most likely be added at some point, in a cleanup pass, as noroadsleft and dunk have been adding them when cleaning up.   

Namely, it's a very useful reference for determining layout support, but this already has a decent amount of layout support, so *shrug*

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**